### PR TITLE
RUST-1553 Add support for document sequences (`OP_MSG` payload type 1)

### PR DESCRIPTION
**Pull request by isabelatkinson**
Adds support for building commands with document sequences (OP_MSG payload type 1) and updates our `Insert` operation to specify the list of documents provided as a document sequence. See the [OP_MSG spec](https://github.com/mongodb/specifications/blob/f1bbc245e68916ab76cb831f6372aa5d6a50ed25/source/message/OP_MSG.rst) for more details.

### CODE REVIEW

File: src/cmap/conn/command.rs [hunk 1]
```diff
@@ -14,37 +13,26 @@ use crate::{
...
+    T: Serialize,
+{
     #[serde(skip)]
     pub(crate) name: String,
 
     #[serde(skip)]
     pub(crate) exhaust_allowed: bool,
 
-    #[serde(flatten)]
+    #[serde(skip)]
```
Comment (abr-egn) [inline src/cmap/conn/command.rs:30]: AFAICT the new behavior (`skip` and `extend_raw_document_buf` in `Message::from_command`) should be equivalent to `flatten`?

File: src/coll/options.rs [hunk 2]
```diff
@@ -151,7 +151,7 @@ pub struct InsertManyOptions {
     pub ordered: Option<bool>,
 
     /// The write concern for the operation.
-    #[serde(skip_deserializing)]
+    #[serde(skip_deserializing, skip_serializing_if = "write_concern_is_empty")]
```
[the remark this answers is not in the document]
Comment (abr-egn) [reply src/coll/options.rs:154]: Sounds good!

**Review by abr-egn — APPROVED** [review]
LGTM!  One clarification question only, no need for re-review.

File: src/cmap/conn/command.rs (hunk 1 above)
Comment (isabelatkinson) [reply src/cmap/conn/command.rs]: Oh yeah you're right, reverted

**Pull request merged**